### PR TITLE
Barchart to fc

### PR DESCRIPTION
**Pull request by ynahmany**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
Refactor BarChart component to Functional component.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: src/BarChart/BarChart.tsx [hunk 1]
```diff
@@ -362,32 +349,19 @@ export class BarChart extends Component<BarChartProps> {
...
+    <ChartContainer
+      id={id}
+      width={width}
+      height={height}
+      margins={margins}
+      xAxisVisible={isAxisVisible(xAxis.props)}
+      yAxisVisible={isAxisVisible(yAxis.props)}
+      className={classNames(css.barChart, className, css[series.props.type])}
+    >
+      {(props) => renderChart(props)}
```
Comment (amcdnl) [inline src/BarChart/BarChart.tsx:364]: I think this could just be this:

```suggestion
      {renderChart}
```

File: src/BarChart/BarChart.tsx [hunk 2]
```diff
@@ -272,27 +266,20 @@ export class BarChart extends Component<BarChartProps> {
...
       domain: axis.props.domain,
       isMultiSeries,
-      isDiverging: this.getIsDiverging()
+      isDiverging: getIsDiverging()
     });
-  }
+  };
 
-  renderChart(containerProps: ChartContainerChildProps) {
+  const renderChart = (containerProps: ChartContainerChildProps) => {
```
Comment (amcdnl) [inline src/BarChart/BarChart.tsx:273]: I think you probably want to do `useCallback` on these for memoization.

File: src/BarChart/BarChart.tsx [hunk 3]
```diff
@@ -244,11 +237,9 @@ export class BarChart extends Component<BarChartProps> {
       groupScale,
       keyScale
     };
-  }
-
-  getKeyScale(data, axis, isMultiSeries: boolean, width: number) {
-    const { series } = this.props;
+  };
 
+  const getKeyScale = (data, axis, isMultiSeries: boolean, width: number) => {
```
Comment (amcdnl) [inline src/BarChart/BarChart.tsx:242]: I need to move the `ChartContainer` to be a hook and then a lot of this code could just be memoized and no need for the `renderChart` method too. Just a side note for later.

**Review by amcdnl — CHANGES_REQUESTED** [review]
A few minor nits but AWESOME overall! 

Just double check all of the storybook stories for bar charts work as expected, these FC changes are a little more sticky than one might expect.

**Comment by amcdnl** [comment]
Any thoughts on the feedback?

File: src/BarChart/StackedNormalizedBarChart.tsx [hunk 4]
```diff
@@ -32,4 +32,8 @@ export const StackedNormalizedBarChart: FC<
     />
   ),
   ...rest
-}) => <BarChart {...rest} series={series} />;
+}) => {
+  console.log('=================');
```
Comment (amcdnl) [inline src/BarChart/StackedNormalizedBarChart.tsx:36]: Need to remove console.

**Comment by amcdnl** [comment]
On the issues w/ stories - this might be helpful - https://github.com/reaviz/reaviz/issues/13 - the parent props item.

**Comment by ynahmany** [comment]
@amcdnl I encounter several issues, so I moved to a new PR, working with some more precise order(starting from a.b.c..).
* keys issue on stories.
* some of the stories was not fully loaded.

**Comment by amcdnl** [comment]
Sounds good - if you are doing a anew pr maybe lets close this?

**Pull request closed without merging**